### PR TITLE
support for crimson packages

### PR DESCRIPTION
**Pull request by ideepika**
This adds a env var OSD_FLAVOR which when set to crimson will also
install ceph-crimson-osd package for CEPH_DEVEL branches.

Signed-off-by: Deepika Upadhyay <dupadhya@redhat.com>

Description of your changes:

Which issue is resolved by this Pull Request:
Resolves #

Checklist:
- [ ] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.

### CODE REVIEW

**Comment by ideepika** [comment]
@dsavineau can you review this PR, thanks!

**Comment by dsavineau** [comment]
> we can now pass `-e BASE_OSD=crimson-osd` for container setup to use
> crimson-osd instead of the default classical OSD.
> This would further serve the purpose for integrating cephadm support for
> crimson.

FYI cephadm doesn't use the ceph/daemon container image but the ceph/ceph container image which only provides the binaries and not the bash scripts

**Comment by ideepika** [comment]
> > we can now pass `-e BASE_OSD=crimson-osd` for container setup to use
> > crimson-osd instead of the default classical OSD.
> > This would further serve the purpose for integrating cephadm support for
> > crimson.
> 
> FYI cephadm doesn't use the ceph/daemon container image but the ceph/ceph container image which only provides the binaries and not the bash scripts
correct me if I am wrong but this commit changed to use daemon-base: https://github.com/ceph/ceph/commit/de448fafe0fe7d6889f3f75ac37f870cde00cf1c

**Comment by dsavineau** [comment]
> > > we can now pass `-e BASE_OSD=crimson-osd` for container setup to use
> > > crimson-osd instead of the default classical OSD.
> > > This would further serve the purpose for integrating cephadm support for
> > > crimson.
> > 
> > 
> > FYI cephadm doesn't use the ceph/daemon container image but the ceph/ceph container image which only provides the binaries and not the bash scripts
> 
> correct me if I am wrong but this commit changed it to use daemon-base: [ceph/ceph@de448fa](https://github.com/ceph/ceph/commit/de448fafe0fe7d6889f3f75ac37f870cde00cf1c)

That's true but only for master branch. Stable branches are using ceph/ceph [1]

So basically ceph/ceph is the same than ceph/daemon-base. But ceph/daemon-base is a binary only container image so you don't have the bash script included.

See https://github.com/ceph/ceph/blob/octopus/doc/install/containers.rst for more information

[1] https://github.com/ceph/ceph/blob/octopus/src/cephadm/cephadm#L3

**Comment by ideepika** [comment]
> > > > we can now pass `-e BASE_OSD=crimson-osd` for container setup to use
> > > > crimson-osd instead of the default classical OSD.
> > > > This would further serve the purpose for integrating cephadm support for
> > > > crimson.
> > > 
> > > 
> > > FYI cephadm doesn't use the ceph/daemon container image but the ceph/ceph container image which only provides the binaries and not the bash scripts
> > 
> > 
> > correct me if I am wrong but this commit changed it to use daemon-base: [ceph/ceph@de448fa](https://github.com/ceph/ceph/commit/de448fafe0fe7d6889f3f75ac37f870cde00cf1c)
> 
> That's true but only for master branch. Stable branches are using ceph/ceph [1]
> 
> So basically ceph/ceph is the same than ceph/daemon-base. But ceph/daemon-base is a binary only container image so you don't have the bash script included.
> 
> See https://github.com/ceph/ceph/blob/octopus/doc/install/containers.rst for more information
> 
> [1] https://github.com/ceph/ceph/blob/octopus/src/cephadm/cephadm#L3

Aah, now I see what you meant. I'll update code to also have bash scripts ( i.e. use ceph/ceph for wip+crimson+centos8 branch pushes.

**Comment by ideepika** [comment]
> > > > we can now pass `-e BASE_OSD=crimson-osd` for container setup to use
> > > > crimson-osd instead of the default classical OSD.
> > > > This would further serve the purpose for integrating cephadm support for
> > > > crimson.
> > > 
> > > 
> > > FYI cephadm doesn't use the ceph/daemon container image but the ceph/ceph container image which only provides the binaries and not the bash scripts
> > 
> > 
> > correct me if I am wrong but this commit changed it to use daemon-base: [ceph/ceph@de448fa](https://github.com/ceph/ceph/commit/de448fafe0fe7d6889f3f75ac37f870cde00cf1c)
> 
> That's true but only for master branch. Stable branches are using ceph/ceph [1]
> 
> So basically ceph/ceph is the same than ceph/daemon-base. But ceph/daemon-base is a binary only container image so you don't have the bash script included.
> 
> See https://github.com/ceph/ceph/blob/octopus/doc/install/containers.rst for more information
> 
> [1] https://github.com/ceph/ceph/blob/octopus/src/cephadm/cephadm#L3

Sure, I agree! But my intention here is to use cephadm for testing crimson PR's , >> ceph-build shall build ceph/daemon container >> will be pushed to quay.io >> add custom image support to cephadm. 

Feel free to correct me, I will test if this is possible locally, will update adding results in a comment. Thanks! :)

**Comment by ideepika** [comment]
@dsavineau I had some issues with testing crimson which took some time to rework, this change will only create a image with crimson packages and does not affect daemon scripts, which is what I want to achieve at this moment.

**Comment by ideepika** [comment]
@dsavineau are the test failures for ceph_ansible-lvm_osds(I don't think seeing at the logs, just neede a confirmation), can you add someone to review/review this, thanks!

**Comment by ideepika** [comment]
@dsavineau maybe this log of my local run can help you review: https://paste.centos.org/view/13897e6d
this pr only add support for installing crimson-osd package and will be used by cephadm for teuthology testing of crimson. 

If you are not available, can someone else help review this PR?

File: contrib/build-push-ceph-container-imgs.sh [hunk 1]
```diff
@@ -34,11 +34,18 @@ fi
 CONTAINER_REPO_USERNAME=${CONTAINER_REPO_USERNAME:-$DOCKER_HUB_USERNAME}
 CONTAINER_REPO_PASSWORD=${CONTAINER_REPO_PASSWORD:-$DOCKER_HUB_PASSWORD}
 
+OUTFILE=../src/daemon-base/__CRIMSON_PACKAGES__
+if [[ "$FLAVOR" == "crimson" ]] && ! grep -q "crimson" $OUTFILE ; then
+  echo "        ceph-crimson-osd__ENV_[CEPH_POINT_RELEASE]__" >> $OUTFILE
+fi
+
```
Comment (dsavineau) [inline contrib/build-push-ceph-container-imgs.sh:41]: We shouldn't do this kind of hack.

This means, this won't work outside of this script whereas one can build the container manually without this script.

Comment (ideepika) [reply contrib/build-push-ceph-container-imgs.sh:41]: sure! sounds good!

**Comment by ideepika** [comment]
@dsavineau I have updated the approach, can you review, thanks in advance.

**Comment by ideepika** [comment]
@dsavineau ping?

**Pull request merged**